src/Fable.Compiler/MSBuildCrackerResolver.fs

```diff
@@ -0,0 +1,179 @@
+namespace Fable.Compiler
+
+open System
+open System.Reflection
+open System.IO
+open System.Diagnostics
+open System.Text.Json
+open Fable
+open Fable.AST
+open Fable.Compiler.ProjectCracker
+
+module private MSBuildCrackerResolver =
+
+    let private fsharpFiles = set [| ".fs"; ".fsi"; ".fsx" |]
+
+    let private isFSharpFile (file: string) =
+        Set.exists (fun (ext: string) -> file.EndsWith(ext, StringComparison.Ordinal)) fsharpFiles
+
+    /// Transform F# files into full paths
+    let private mkOptionsFullPath (projectFile: FileInfo) (compilerArgs: string array) : string array =
+        compilerArgs
+        |> Array.map (fun (line: string) ->
+            if not (isFSharpFile line) then
+                line
+            else
+                Path.Combine(projectFile.DirectoryName, line)
+        )
+
+    type FullPath = string
+
+    let private dotnet_msbuild_with_defines (fsproj: FullPath) (args: string) (defines: string list) : Async<string> =
+        backgroundTask {
+            let psi = ProcessStartInfo "dotnet"
+            let pwd = Assembly.GetEntryAssembly().Location |> Path.GetDirectoryName
+
+            psi.WorkingDirectory <- Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)
+
+            psi.Arguments <- $"msbuild \"%s{fsproj}\" %s{args}"
+            psi.RedirectStandardOutput <- true
+            psi.RedirectStandardError <- true
+            psi.UseShellExecute <- false
+
+            if not (List.isEmpty defines) then
+                let definesValue = defines |> String.concat ";"
+                psi.Environment.Add("DefineConstants", definesValue)
+
+            use ps = new Process()
+            ps.StartInfo <- psi
+            ps.Start() |> ignore
+            let output = ps.StandardOutput.ReadToEnd()
+            let error = ps.StandardError.ReadToEnd()
+            do! ps.WaitForExitAsync()
+
+            let fullCommand = $"dotnet msbuild %s{fsproj} %s{args}"
+            // printfn "%s" fullCommand
+
+            if not (String.IsNullOrWhiteSpace error) then
+                failwithf $"In %s{pwd}:\n%s{fullCommand}\nfailed with\n%s{error}"
+
+            return output.Trim()
+        }
+        |> Async.AwaitTask
+
+    let private dotnet_msbuild (fsproj: FullPath) (args: string) : Async<string> =
+        dotnet_msbuild_with_defines fsproj args List.empty
+
+    let mkOptionsFromDesignTimeBuildAux (fsproj: FileInfo) (options: CrackerOptions) : Async<ProjectOptionsResponse> =
+        async {
+            let! targetFrameworkJson =
+                let configuration =
+                    if String.IsNullOrWhiteSpace options.Configuration then
+                        ""
+                    else
+                        $"/p:Configuration=%s{options.Configuration} "
```

Comment:
We can move this variable in the scope above as we have a similar logic on line 102-103.